_ml-commons-plugin/agents-tools/tools/connector-tool.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: default
+title: Connector tool
+has_children: false
+has_toc: false
+nav_order: 20
+parent: Tools
+grand_parent: Agents and tools
+---
+
+<!-- vale off -->
+# Connector tool
+**Introduced 2.15**
+{: .label .label-purple }
+<!-- vale on -->
+
+The `ConnectorTool` runs `execute` action in a connector.
+
+## Step 1: Register a connector with execute action
+
+`ConnectorTool` can only run `execute` action in a connector. So you need to create a connector with `execute` action. It's very similar to the `predict` action in connector. For example, create a connector to execute an AWS Lambda function with [Function URL](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html).
+The Lambda function accepts two integer numbers and returns the sum.
+
+```
+POST _plugins/_ml/connectors/_create
+{
+  "name": "Lambda connector of simple calculator",
+  "description": "Demo connector of lambda function",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "YOUR AWS REGION",
+    "service_name": "lambda"
+  },
+  "credential": {
+    "access_key": "YOUR ACCESS KEY",
+    "secret_key": "YOUR SECRET KEY",
+    "session_token": "YOUR SESSION TOKEN"
+  },
+  "actions": [
+    {
+      "action_type": "execute",
+      "method": "POST",
+      "url": "YOUR LAMBDA FUNCTION URL",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "request_body": "{ \"number1\":\"${parameters.number1}\", \"number2\":\"${parameters.number2}\" }"
+    }
+  ]
+}
+```
+OpenSearch responds with an agent ID:
+
+```json
+{
+  "connector_id": "Zz1XEJABXWrLmr4mewEF"
+}
+```
+
+## Step 2: Register a flow agent that will run the ConnectorTool
+
+A flow agent runs a sequence of tools in order and returns the last tool's output. To create a flow agent, send the following register agent request:
+
+Suppose the Lambda function will return such response
+```json
+{
+  "result": 10
+}
+```
+Need to set `response_filter` to retrieve the result `10` with json path `$.result`.
```

Comment:
maybe try explain a bit more why we need response_filter? If we don't use response_filter, what happens? is it also include the `"name": "response"`?

Reply:
Add more details